torch_np/_wrapper.py

```diff
@@ -497,10 +492,35 @@ def ravel_multi_index(multi_index, dims, mode='raise', order='C'):
     return sum(idx*dim for idx, dim in zip(multi_index, dims))
 
 
-# YYY : pattern: array_like input, tuple of arrays as output; cf broadcast_arrays
 def nonzero(a):
-    a = asarray(a).get()
-    return tuple(asarray(_) for _ in a.nonzero(as_tuple=True))
+    arr = asarray(a)
+    return arr.nonzero()
+
+
+def flatnonzero(a):
+    arr = asarray(a)
+    return nonzero(arr.ravel())[0]
+
+
+def argwhere(a):
+    arr = asarray(a)
+    if arr.ndim == 0:
+        a = atleast_1d(a)
+        # then remove the added dimension
+        return argwhere(a)[:, :0]
+    return transpose(nonzero(a))
```

Comment:
The strides of this won't match those of NumPy for sure, but that's alright, let's not care about that for now. Now, it would be good to leave a note in the list of design decisions.

Reply:
Not sure what to note, could you elaborate? For zero-dim inputs strides seem consistent in pytorch at least (as far as strides for zero-dim arrays are actually relevant, but that's a different story)

```
In [204]: torch.argwhere(torch.empty((2,)*0, dtype=torch.int32)).stride()
Out[204]: (1, 1)

In [205]: torch.argwhere(torch.empty((2,)*0, dtype=torch.int64)).stride()
Out[205]: (1, 1)

In [206]: torch.argwhere(torch.empty((2,)*0, dtype=torch.bool)).stride()
Out[206]: (1, 1)

In [207]: _np.argwhere(_np.empty((2,)*0, dtype='int32')).strides
Out[207]: (8, 8)

In [208]: _np.argwhere(_np.empty((2,)*0, dtype='int64')).strides
Out[208]: (0, 0)

In [209]: _np.argwhere(_np.empty((2,)*0, dtype='bool')).strides
Out[209]: (8, 8)
```

Reply:
For larger inputs, the strides returned by this implementation will be transposed, while I'd expect those from NumPy to be contiguous.

Reply:
OK,  with `argwhere` just delegating to `pytorch.argwhere`, it all seems consistent (modulo byte vs element strides):

```
In [133]: t
Out[133]: 
tensor([[0., 1., 2., 3., 4.],
        [5., 6., 7., 8., 9.]])

In [134]: w.asarray(t)
Out[134]: 
array_w([[0., 1., 2., 3., 4.],
        [5., 6., 7., 8., 9.]])

In [135]: w.argwhere(w.asarray(t))
Out[135]: 
array_w([[0, 1],
        [0, 2],
        [0, 3],
        [0, 4],
        [1, 0],
        [1, 1],
        [1, 2],
        [1, 3],
        [1, 4]])

In [136]: w.argwhere(w.asarray(t)).strides
Out[136]: (1, 9)

In [137]: np.argwhere(t.numpy()).strides
Out[137]: (8, 72)
```